command/server/config.go

```diff
@@ -236,6 +239,14 @@ type GRPCConfig struct {
 	Addr string
 }
 
+type PrometheusConfig struct {
+	// Enabled selects whether the api is enabled
+	Enabled bool `hcl:"enabled,optional"`
+
+	// Host is the address to bind the api
+	address string `hcl:"host,optional"`
```

Comment:
Several things:
- For simplicity, use the same name for the var as the HCL/Json name in the tag. So this one should be called host.
- The variable should be in upper case like "Enabled". Otherwise, it is not public and it will not be set during config decoding.
- Lets use only one variable PrometheusAddr. If it is different from zero it is enabled.

Reply:
Also, please add a flag to set this field as well.